addon/components/bs-form/element/control/switch.hbs

```diff
@@ -0,0 +1,11 @@
+<input
+  type="checkbox"
+  id={{@id}}
+  disabled={{this.disabled}}
+  readonly={{this.readonly}}
+  aria-describedby={{this.ariaDescribedBy}}
+  checked={{this.value}}
+  class="custom-control-input {{this.formValidationClass}}"
+  ...attributes
+  {{on "click" this.handleClick}}
+>
```

Comment:
Wondering if it makes sense to refactor to:

```hbs
<BsForm::Element::Control::Checkbox
  @value={{@value}}
  @id={{@id}}
  @type="checkbox"
  @label={{@label}}
  @disabled={{@disabled}}
  @readonly={{@readonly}}
  @required={{@required}}
  @options={{@options}}
  @optionLabelPath={{@optionLabelPath}}
  @ariaDescribedBy={{@ariaDescribedBy}}
  @onChange={{@onChange}}
  @validationType={{@validationType}}
  @size={{@size}}
  class="custom-control-input"
  ...attributes
/>
```

We would avoid some code duplication. But needing to explicitly pass down all arguments is a big trade-off. We could only passthrough the arguments actually used by `<BsForm::Element::Control::Checkbox />`. But that would be also error prune...

So not a request to change. More a question on long-term architectural patterns for @simonihmig. :smile_cat:

Reply:
The problem is that the checkbox control adds a `form-check-input` class, which we do not want on the switch checkbox. Which is why I went with duplicating the control.

Reply:
Missed that one. Thanks for explaining.